# Default parameters for the demo datasets
# Documentation of the paramters can be found in the corresponding reconfigure files (see <package_name>/cfg/<package_name>/*.py)
# running rqt_reconfigure also provides tooltip text for the parameters and allows for online tuning

# General
frame_id: "odom"

# Publishers
publisher_rate: 30
pose_path_publisher_rate: 5

# Subscribers
topic_imu: "/sensor_platform/imu"
topic_baro_altimeter: "/sensor_platform/baro"
topic_radar_scan: "/sensor_platform/radar/scan"
topic_radar_trigger: "/sensor_platform/radar/trigger"

# Rosbag mode --> used by rosbag node
republish_ground_truth: True
topic_ground_truth_pose: "/ground_truth/pose"
topic_ground_truth_twist: "/ground_truth/twist"
topic_ground_truth_twist_body: "/ground_truth/twist_body"

# KF Updates
altimeter_update: False
radar_update: True
radar_yaw_update: False

sigma_altimeter: 5.0
sigma_b_alt: 0

# Radar Measurement Model
outlier_percentil_radar: 0.01
use_w: True

# Radar Ego Velocity Estimation
min_dist: 0.5
max_dist: 100
min_db: 3
elevation_thresh_deg: 45
azimuth_thresh_deg: 60
radar_velocity_correction_factor: 1.0

thresh_median_zero_velocity: 0.05
allowed_outlier_percentage: 0.5
sigma_zero_velocity_x: 0.025 # right
sigma_zero_velocity_y: 0.025 # forward
sigma_zero_velocity_z: 0.025 # up

sigma_offset_radar_x: 0.075   # right
sigma_offset_radar_y: 0.05  # forward
sigma_offset_radar_z: 0.075  # up

max_sigma_x: 0.2   # right
max_sigma_y: 0.15  # forward
max_sigma_z: 0.2   # up
max_r_cond: 1.0e3
use_cholesky_instead_of_bdcsvd: True # faster but less stable

use_ransac: True
outlier_prob: 0.5
success_prob: 0.999
N_ransac_points: 3
inlier_thresh: 0.15

# ODR refinement
use_odr: True    # turn on odr refinement
min_speed_odr: 4.0 # min speed for ODR refinement
sigma_v_r: 0.125  # noise of v_r measurement used for the refinement
model_noise_offset_deg: 2.0 # min model noise
model_noise_scale_deg: 10.0 # scale model noise

# Initialization
T_init: 10
calib_gyro: true
g_n: 9.81

p_0_x: 0
p_0_y: 0
p_0_z: 0

v_0_x: 0
v_0_y: 0
v_0_z: 0

yaw_0_deg: 0

b_0_a_x: 0
b_0_a_y: 0
b_0_a_z: 0

b_0_w_x_deg: 0
b_0_w_y_deg: 0
b_0_w_z_deg: 0

b_0_alt: 0

# Initial Uncertainty
sigma_p: 0
sigma_v: 0
sigma_roll_pitch_deg: 1.5
sigma_yaw_deg: 0
sigma_b_a: 0.1
sigma_b_w_deg: 0.1

sigma_l_b_r_x: 0.02
sigma_l_b_r_y: 0.02
sigma_l_b_r_z: 0.02
sigma_eul_b_r_roll_deg: 1.5
sigma_eul_b_r_pitch_deg: 1.5
sigma_eul_b_r_yaw_deg: 1.5

# Noise PSDs
noise_psd_a: 0.2
noise_psd_w_deg: 0.25
noise_psd_b_a: 0.00001
noise_psd_b_w_deg: 0.00001
noise_psd_b_alt: 0.0001

# Sim noise
sim_sigma_acc: 0.33
sim_sigma_gyro: 0.03
sim_sigma_v_r_x: 0.05
sim_sigma_v_r_y: 0.05
sim_sigma_v_r_z: 0.05
